### PR TITLE
[Mailgun] Send Attachments With Email

### LINKED ISSUE #3901: [FEATURE] Mailgun SendEmail support attachments
**Describe the solution you'd like**
Add support for adding attachments for SendEmail

### PR DESCRIPTION
**Pull request by andrewjschuang**
Resolves #3901.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4422"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Review by feyzullah — COMMENTED** [review]
Hi @andrewjschuang, I left just 2 commits about used packages. Can you please check?

File: components/mailgun/package.json [hunk 1]
```diff
@@ -14,8 +14,12 @@
     "access": "public"
   },
   "dependencies": {
+    "@pipedream/helper_functions": "^0.3.7",
+    "@pipedream/platform": "^1.1.1",
     "form-data": "^4.0.0",
+    "lodash": "^4.17.21",
     "lodash.get": "^4.4.2",
+    "mailgun-js": "^0.22.0",
```
[the remark this answers is not in the document]
Comment (andrewjschuang) [reply components/mailgun/package.json]: Ok, I removed the deprecated `mailgun-js` package and created attachments manually! I didn't find a way to do it with the `mailgun.js` package.

File: components/mailgun/package.json [hunk 2]
```diff
@@ -14,8 +14,12 @@
...
     "mailgun.js": "^3.5.2"
```
[the remark this answers is not in the document]
Comment (andrewjschuang) [reply components/mailgun/package.json:22]: I'm not sure, but I don't think it is _necessary_ to update to the latest version. Would that be ok?

[the remark this answers is not in the document]
Comment (feyzullah) [reply components/mailgun/package.json:22]: Sure, it was optional and to keep up with the current lib. There are multiple components already, so if we are not sure, we can definitely keep it as it was.

**Review by feyzullah — APPROVED** [review]

**Comment by DilanAthukorala** [comment]
Verification completed, ready for release.

**Pull request merged**